packages/xo-web/src/xo-app/disk-import/index.js

```diff
@@ -0,0 +1,188 @@
+import _, { messages } from 'intl'
+import ActionButton from 'action-button'
+import Button from 'button'
+import Collapse from 'collapse'
+import decorate from 'apply-decorators'
+import Dropzone from 'dropzone'
+import fromEvent from 'promise-toolbox/fromEvent'
+import React, { Component } from 'react'
+import { formatSize, mapPlus } from 'utils'
+import { generateId, linkState } from 'reaclette-utils'
+import { importDisks } from 'xo'
+import { injectIntl } from 'react-intl'
+import { injectState, provideState } from 'reaclette'
+import { LabelCol, InputCol, Row } from 'form-grid'
+import { readCapacityAndGrainTable } from 'xo-vmdk-to-vhd'
+import { SelectSr } from 'select-objects'
+
+const getInitialState = () => ({
+  disks: [],
+  mapDescriptions: {},
+  mapNames: {},
+  sr: undefined,
+})
+
+const DiskImport = decorate([
+  provideState({
+    initialState: getInitialState,
+    effects: {
+      handleDrop: (effects, files) => async ({ sr }) => {
+        const disks = await Promise.all(
+          mapPlus(files, async (file, push) => {
+            const { name } = file
+            const extIndex = name.lastIndexOf('.')
+            let type
+            if (
+              extIndex >= 0 &&
+              (type = name.slice(extIndex + 1)) &&
+              (type === 'vmdk' || type === 'vhd')
+            ) {
+              let vmdkData
+              if (type === 'vmdk') {
+                const parsed = await effects.parseVmdk(file)
+                vmdkData = {
+                  blocksTable: await parsed.tablePromise,
+                  capacity: parsed.capacityBytes,
+                }
+              }
+              push({
+                id: generateId(),
+                file,
+                name,
+                sr,
+                type,
+                vmdkData,
+              })
+            }
+          })
+        )
+        return { disks }
+      },
+      linkState,
+      onChangeDescription: (_, { target: { name, value } }) => ({
+        mapDescriptions,
+      }) => {
+        mapDescriptions[name] = value
+        return { mapDescriptions }
+      },
+      onChangeName: (_, { target: { name, value } }) => ({ mapNames }) => {
+        mapNames[name] = value
+        return { mapNames }
+      },
+      onChangeSr: (_, sr) => ({ sr }),
```

Comment:
`linkState`?

Reply:
`linkState` get value from `event.target` and here we use `SelectSr` that use `react-select`.